qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/VisitTypeSteps.java

```diff
@@ -0,0 +1,104 @@
+import io.cucumber.java.en.When;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.openmrs.contrib.qaframework.RunTest;
+import org.openmrs.contrib.qaframework.helper.RestClient;
+import org.openmrs.contrib.qaframework.page.AdministrationPage;
+import org.openmrs.contrib.qaframework.page.VisitTypeListPage;
+import org.openmrs.contrib.qaframework.page.VisitTypePage;
+
+import static org.junit.Assert.assertTrue;
+
+public class VisitTypeSteps extends Steps {
+
+    private AdministrationPage administrationPage;
+    private VisitTypePage visitTypePage;
+    private VisitTypeListPage visitTypeListPage;
+    private String visitTypeUuid;
+    private final String VISIT_TYPE_NAME = RandomStringUtils.randomAlphabetic(5);
+    private static final String VISIT_TYPE_DESCRIPTION = "Description for Visit Type test workflow";
+
+    @Before(RunTest.HOOK.SELENIUM_VISIT_TYPE)
+    public void setup() {
+        initiateWithLogin();
+    }
+
+    @After(RunTest.HOOK.SELENIUM_VISIT_TYPE)
+    public void teardown() {
+        quit();
+        RestClient.delete("visittype/" + visitTypeUuid, true);
+    }
+
+    @Given("User clicks the system administration link on the home page")
+    public void launchSystemAdministrationPage() {
+        administrationPage = homePage.goToAdministration();
+    }
+
+    @When("User clicks the manage Visit Types on the administration page")
+    public void launchAdvancedAdministrationPage() {
+        visitTypeListPage = administrationPage.goToVisitTypePage();
+    }
+
+    @And("User clicks the add visit type link on the manage visit type page")
+    public void clickOnAddVisitTypeLink() {
+        visitTypePage = visitTypeListPage.addVisitType();
+    }
+
+    @And("User fills the add visit type form")
+    public void fillAddVisitTypeForm(){
+        visitTypePage.setName(VISIT_TYPE_NAME);
+        visitTypePage.setDescription(VISIT_TYPE_DESCRIPTION);
+    }
+
+    @And ("User saves visit type")
+    public void saveVisitType() {
+        visitTypePage.save();
+    }
+
+    @Then("visit type is saved successfully")
+    public void systemAddsVisitType() {
+        assertTrue(textExists("Visit Type saved"));
+    }
+
+    @And("User clicks on already saved visit type")
+    public void clickAlreadySavedVisitType() {
+        visitTypePage = visitTypeListPage.clickAlreadySavedVisitType();
+    }
+
+    @And("User edits the visit type form")
+    public void editVisitType(){
+        visitTypePage.setName("new VisitTypeName");
+        visitTypePage.save();
+    }
+
+    @And ("User fills in the reason for retiring visit type")
+    public void setRetireReason() {
+        visitTypePage.setRetireReason("retire reason");
+    }
+
+    @And("User clicks on retire visit type button")
+    public void retireVisitType(){
+        visitTypePage.retire();
+    }
+
+    @Then ("User clicks on delete visit type forever button")
+    public void deleteVisitTypeForever() {
+        visitTypePage.delete();
+    }
+
```

Comment:
Ensure to add one single line at the end of each class to cater for git.

Reply:
removed it